packages/react-dom-bindings/src/shared/ReactDOMResources.js

```diff
@@ -0,0 +1,107 @@
+/**
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+
+import type {ReactNodeList} from 'shared/ReactTypes';
+
+import isArray from 'shared/isArray';
+import {
+  REACT_ELEMENT_TYPE,
+  REACT_FORWARD_REF_TYPE,
+  REACT_FRAGMENT_TYPE,
+} from 'shared/ReactSymbols';
+
+export function concatTextChildrenProd(children: ReactNodeList): string {
```

Comment:
This file is being extracted because I intend to move all of the common logic such as key generation to a shared file. This will help ensure that Fizz and the client always agree on what is and is not a valid resource identity. For now the only thing migrated here is this helper function